process/block/sovereignChainBlock_test.go

```diff
@@ -883,3 +893,269 @@ func TestSovereignShardProcessor_CreateBlock(t *testing.T) {
+		require.Nil(t, err)
+
+		_, _, err = scbp.ProcessBlock(&block.SovereignChainHeader{}, nil, haveTime)
+		require.Equal(t, process.ErrNilBlockBody, err)
+	})
+	t.Run("not enough time should error", func(t *testing.T) {
+		sovArgs := createSovChainBlockProcessorArgs()
+		scbp, err := blproc.NewSovereignChainBlockProcessor(sovArgs)
+		require.Nil(t, err)
+
+		_, _, err = scbp.ProcessBlock(nil, nil, nil)
+		require.Equal(t, process.ErrNilHaveTimeHandler, err)
+	})
+	t.Run("process header with incorrect epoch should error", func(t *testing.T) {
+		randSeed := []byte("rand seed")
+		blkc, _ := blockchain.NewBlockChain(&statusHandlerMock.AppStatusHandlerStub{})
+		_ = blkc.SetCurrentBlockHeaderAndRootHash(
+			&block.Header{
+				Nonce:    0,
+				Round:    0,
+				Epoch:    1,
+				RandSeed: randSeed,
+			}, []byte("root hash"),
+		)
+		_ = blkc.SetGenesisHeader(&block.Header{Nonce: 0})
+		blkc.SetCurrentBlockHeaderHash([]byte("zzz"))
+
+		coreComponents, dataComponents, bootstrapComponents, statusComponents := createComponentHolderMocks()
+		dataComponents.BlockChain = blkc
+		arguments := createSovChainBaseBlockProcessorArgs(coreComponents, dataComponents, bootstrapComponents, statusComponents)
+		sovArgs := createArgsSovereignChainBlockProcessor(arguments)
+		scbp, err := blproc.NewSovereignChainBlockProcessor(sovArgs)
+		require.Nil(t, err)
+
+		hdr := &block.Header{
+			Nonce:         1,
+			Round:         1,
+			Epoch:         0,
+			PubKeysBitmap: []byte("0100101"),
+			PrevHash:      []byte("zzz"),
+			PrevRandSeed:  randSeed,
+			Signature:     []byte("signature"),
+			RootHash:      []byte("root hash"),
+		}
+		_, _, err = scbp.ProcessBlock(hdr, &block.Body{}, haveTime)
+		require.Equal(t, process.ErrEpochDoesNotMatch, err)
+	})
+	t.Run("create block started should error because add intermediate txs in tx coordinator returns error", func(t *testing.T) {
+		expectedErr := fmt.Errorf("createBlockStarted error")
+
+		coreComponents, dataComponents, bootstrapComponents, statusComponents := createComponentHolderMocks()
+		arguments := createSovChainBaseBlockProcessorArgs(coreComponents, dataComponents, bootstrapComponents, statusComponents)
+		arguments.TxCoordinator = &testscommon.TransactionCoordinatorMock{
+			AddIntermediateTransactionsCalled: func(_ map[block.Type][]data.TransactionHandler, _ []byte) error {
+				return expectedErr
+			},
+		}
+		sovArgs := createArgsSovereignChainBlockProcessor(arguments)
+
+		scbp, err := blproc.NewSovereignChainBlockProcessor(sovArgs)
+		require.Nil(t, err)
+
+		expectedSovHeader := &block.SovereignChainHeader{
+			Header: &block.Header{
+				Nonce: 1,
+				Round: 1,
+				Epoch: 0,
+			},
+		}
+
+		hdr, bodyHandler, err := scbp.ProcessBlock(expectedSovHeader, &block.Body{}, haveTime)
+		require.True(t, check.IfNil(bodyHandler))
+		require.True(t, check.IfNil(hdr))
+		require.Equal(t, expectedErr, err)
+	})
+	t.Run("data not prepared for processing should error should error", func(t *testing.T) {
```

Comment:
should double error :laughing: 
```suggestion
	t.Run("data not prepared for processing should error", func(t *testing.T) {
```